app/networkapi/utility/decorators.py

```diff
@@ -0,0 +1,45 @@
+import logging
+
+from threading import Timer
+from django.utils import timezone
+
+logger = logging.getLogger(__name__)
+
+
+def debounce_and_throttle(debounce_seconds, throttle_seconds):
+    """
+    A Decorator for debouncing and throttling a function's execution
+    the original debounce strategy was found here:
+    https://gist.github.com/walkermatt/2871026
+    """
+    def decorator(fn):
+        def debounced_and_throttled(*args, **kwargs):
+            def call_fn():
+                now = timezone.now()
+                try:
+                    time_delta = now - debounced_and_throttled.last_called
+                    if time_delta.seconds < throttle_seconds:
+                        return logger.info(
```

Comment:
this looks like it would log this for every debounce - does that make sense? I don't know if this will lead to log bloat (I suspect not, but you never know)